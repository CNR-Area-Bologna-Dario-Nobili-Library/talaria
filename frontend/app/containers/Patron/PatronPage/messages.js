/*
 * PatronPage Messages
 *
 * This contains all the text for the PatronPage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.PatronPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Hi I\'m the unuseful message..!',
  },
});
