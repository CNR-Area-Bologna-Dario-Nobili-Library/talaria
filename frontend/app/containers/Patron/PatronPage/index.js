/*
 * PatronPage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React, { useEffect } from 'react';
import { FormattedMessage } from 'react-intl';
import messages from './messages';
import {createStructuredSelector} from "reselect";
import {compose} from "redux";
import { connect } from 'react-redux';
import {BasePage} from "components";
import patronRoutes from "routes/patronRoutes";
import makeSelectPatron  from '../selectors';
import {requestGetLibraryList} from '../actions'

function PatronPage(props) {
  console.log('PatronPage', props)

  const {patron, dispatch} = props
  
  useEffect(() => {
    if(!patron.loading) {
      dispatch(requestGetLibraryList())
    }
   }, [])

  return (
    <>
      <BasePage {...props} routes={patronRoutes} messages={messages} />
    </>
  );
}
const mapStateToProps = createStructuredSelector({
  patron: makeSelectPatron()
});

const mapDispatchToProps = (dispatch) => ({
  dispatch,
})

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

// export default withRouter(withGoogleReCaptcha((SignupForm)));
export default compose(withConnect)(PatronPage);
