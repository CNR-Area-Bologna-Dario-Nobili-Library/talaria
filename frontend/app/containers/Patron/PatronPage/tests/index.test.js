import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import PatronPage from '../index';

describe('<PatronPage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <PatronPage />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
