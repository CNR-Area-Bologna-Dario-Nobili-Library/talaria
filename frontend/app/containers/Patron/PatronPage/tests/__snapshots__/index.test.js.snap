// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PatronPage /> should render and match the snapshot 1`] = `
<h1>
  <span>
    This is the PatronPage container!
  </span>
</h1>
`;
